pkg/trace/stats/concentrator.go

```diff
@@ -215,6 +228,59 @@ func (c *Concentrator) Add(t Input) {
 	c.mu.Unlock()
 }
 
+// ProcessOTLPTraces applies APM stats calculation on the otlp traces in the concentrator.
+// This function is NOT called in Concentrator.Run(), so you need to manually call it to get APM stats on the given OTLP traces.
+func (c *Concentrator) ProcessOTLPTraces(traces ptrace.Traces, additionalCtagKeys ...string) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	includeCID := shouldIncludeCIDDim(c.conf)
+	spanByID, resByID, scopeByID := traceutil.IndexOTelSpans(traces)
+	topLevelByKind := c.conf.HasFeature("enable_otlp_compute_top_level_by_span_kind")
+	topLevelSpans := traceutil.GetTopLevelOTelSpans(spanByID, resByID, topLevelByKind)
+	ignoreResNames := make(map[string]struct{})
+	for _, resName := range c.conf.Ignore["resource"] {
+		ignoreResNames[resName] = struct{}{}
+	}
+	for spanID, span := range spanByID {
+		res := resByID[spanID]
+		resName := traceutil.GetOTelResource(span, res)
+		if _, exists := ignoreResNames[resName]; exists {
+			continue
+		}
+		env := traceutil.GetOTelAttrValInResAndSpanAttrs(span, res, true, semconv.AttributeDeploymentEnvironment)
+		if env == "" {
+			env = c.agentEnv
+		}
+		aggKey := PayloadAggregationKey{
+			Env:      env,
+			Hostname: traceutil.GetOTelHostname(span, res, c.conf.OTLPReceiver.AttributesTranslator, c.conf.Hostname),
+			Version:  traceutil.GetOTelAttrValInResAndSpanAttrs(span, res, true, semconv.AttributeServiceVersion),
+		}
+		if includeCID {
+			if cid := traceutil.GetOTelAttrValInResAndSpanAttrs(span, res, true, semconv.AttributeContainerID, semconv.AttributeK8SPodUID); cid != "" {
+				aggKey.ContainerID = cid
+				if gitCommitSha, imageTag, err := version.GetVersionDataFromContainerTags(cid, c.conf); err == nil {
+					aggKey.GitCommitSha = gitCommitSha
+					aggKey.ImageTag = imageTag
+				}
```

Comment:
I believe these fields which were very recently introduced are in the process of being rolled back.

Reply:
That's good to know, I removed these fields.